SROS2_Linux.md

```diff
@@ -85,13 +85,12 @@ These variables need to be defined in each terminal used for the demo. For conve
 
 ### Run the demo
 
-ROS2 allows you to [change DDS implementation at runtime](https://docs.ros.org/en/rolling/Guides/Working-with-multiple-RMW-implementations.html).
-This demo can be run with fastrtps by setting:
-```bash
-export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
-```
-And with Connext by setting:
+ROS 2 allows you to [change DDS implementation at runtime](https://docs.ros.org/en/rolling/Guides/Working-with-multiple-RMW-implementations.html).
+This demo can be run with FastDDS / CycloneDDS / ConnextDDS by setting the `RMW_IMPLEMENTATION` variable, e.g.:
+
 ```bash
+export RMW_IMPLEMENTATION=rmw_fastrtps_cpp  # or
+export RMW_IMPLEMENTATION=rmw_cyclonedds_cpp  # or
```

Comment:
with `rmw_cyclonedds_cpp`, no problem observed.

```bash
root@51cdd59e1f3e:~# export RMW_IMPLEMENTATION=rmw_cyclonedds_cpp
root@51cdd59e1f3e:~# export ROS_SECURITY_KEYSTORE=~/sros2_demo/demo_keystore
root@51cdd59e1f3e:~# export ROS_SECURITY_ENABLE=true
root@51cdd59e1f3e:~# export ROS_SECURITY_STRATEGY=Enforce
root@51cdd59e1f3e:~# ros2 run demo_nodes_cpp talker --ros-args --enclave /talker_listener/talker
[INFO] [1715030073.646233306] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/talker
[INFO] [1715030074.656773123] [talker]: Publishing: 'Hello World: 1'
...<snip>

root@51cdd59e1f3e:~# export RMW_IMPLEMENTATION=rmw_cyclonedds_cpp
root@51cdd59e1f3e:~# export ROS_SECURITY_KEYSTORE=~/sros2_demo/demo_keystore
root@51cdd59e1f3e:~# export ROS_SECURITY_ENABLE=true
root@51cdd59e1f3e:~# export ROS_SECURITY_STRATEGY=Enforce
root@51cdd59e1f3e:~# ros2 run demo_nodes_py listener --ros-args --enclave /talker_listener/listener
[INFO] [1715030308.304379343] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/listener
[INFO] [1715030308.870311399] [listener]: I heard: [Hello World: 29]

root@51cdd59e1f3e:~# export ROS_SECURITY_ENCLAVE_OVERRIDE=/talker_listener/listener
root@51cdd59e1f3e:~# ros2 daemon stop
root@51cdd59e1f3e:~# ros2 daemon status
root@51cdd59e1f3e:~# ros2 node list --no-daemon
[INFO] [1715030359.385181616] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/listener
/talker
root@51cdd59e1f3e:~# ros2 topic list --no-daemon
[INFO] [1715030366.263345203] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/listener
/chatter
/parameter_events
/rosout
root@51cdd59e1f3e:~# ros2 topic echo /chatter
[INFO] [1715030372.966397551] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/listener
data: 'Hello World: 94'
---
data: 'Hello World: 95'
---
```

Reply:
Ah good as that's the only one I tested end-to-end from https://github.com/osrf/ros2_test_cases/issues/1307